script/preview.js

```diff
@@ -53,62 +59,120 @@ if (options.buildpath === null) {
 const app = express();
 const drupalClient = getDrupalClient(options);
 
-app.use(express.static(path.join(__dirname, '..', options.buildpath)));
+const urls = {
+  [ENVIRONMENTS.LOCALHOST]: 'http://localhost:3001',
+  [ENVIRONMENTS.VAGOVDEV]:
+    'http://dev.va.gov.s3-website-us-gov-west-1.amazonaws.com',
+  [ENVIRONMENTS.VAGOVSTAGING]:
+    'http://staging.va.gov.s3-website-us-gov-west-1.amazonaws.com',
+  [ENVIRONMENTS.VAGOVPROD]:
+    'http://www.va.gov.s3-website-us-gov-west-1.amazonaws.com',
+};
 
 app.get('/health', (req, res) => {
   res.send(200);
 });
 
-app.get('/preview', async (req, res) => {
-  const smith = createPipieline({
-    ...options,
-    port: process.env.PORT || 3001,
-  });
+app.get('/preview', async (req, res, next) => {
+  try {
+    const smith = createPipieline({
+      ...options,
+      port: process.env.PORT || 3001,
+    });
+
+    const requests = [
+      drupalClient.getLatestPageById(req.query.nodeId),
+      fetch(`${urls[options.buildtype]}/generated/file-manifest.json`).then(
+        resp => {
+          if (resp.ok) {
+            return resp.json();
+          }
+          throw new Error(
+            `HTTP error when fetching manifest: ${resp.status} ${
+              resp.statusText
+            }`,
+          );
+        },
+      ),
+      fetch(
+        `${urls[options.buildtype]}/generated/drupalHeaderFooter.json`,
+      ).then(resp => {
+        if (resp.ok) {
+          return resp.json();
+        }
+        throw new Error(
+          `HTTP error when fetching header/footer data: ${resp.status} ${
+            resp.statusText
+          }`,
+        );
+      }),
+    ];
+
+    const [drupalData, fileManifest, headerFooterData] = await Promise.all(
```

Comment:
These are done sort of in parallel, so hopefully that helps with the potential performance issues